### DISCUSSION TITLE
Support custom variants

### DISCUSSION THREAD

**Discussion by adamgen** [ben-rogerson/twin.macro#416] [Ideas]
When adding a plugin that adds new variants it isn't added to the macro by default. How can I approach this? Am I missing some config or should I jump to the source code?

**Comment by adamgen** [comment]
I see now that it's not as simple... And there's no api plugin for the macro. Have we considered adding a plugin API? Or about an implementation for using tailwind.config somehow?

**Comment by ben-rogerson** [comment]
Yeah it's not a simple solution - twin will need a custom solution for this.

[Check out the add variant discussion for more →](https://github.com/ben-rogerson/twin.macro/discussions/320)
